### ISSUE TITLE
How to add multiple Prophet forecasts?

### ISSUE THREAD

**Issue by bernardo-suez** [facebook/prophet#2126]
I have one time series of water consumption for each consumer type (residential, commercial, etc.) and have to use them to make demand forecasts. I wanted to just add all series and run one forecast of combined consumption, but my boss wants me to do one forecast by consumer type and only after combine them into one forecast---the stakeholders may choose different assumptions for each class. How do I go about calculating the combined uncertainty interval? 

I have tried using the usual square root of the summation of variances plus covariances (I have `interval_width=0.95`, which amounts to two standard deviations), but (a) that returned a much wider uncertainty interval than it makes sense (the summation of all consumption means falling entirely well within two standard deviations) and (b) the uncertainty intervals in Prophet are not simetrical, so just having the standard deviation wouldn't do. I thought of using the samples within each model but then remembered they wouldn't be correlated. Any other ideas?

**Comment by ruben-rosa** [comment]
This is not an easy answer here. It will depend on the business assumptions. 

There are advantages and disadvantages of doing multi-level forecast. 

1) Top - Down approach
2) Bottom - Up approach
2) Middle approach 

If you have a 3 level TS then:

Option 1: Using option 1 you work on level 1 and using proportions calculate level 2 and level 3. Easy to do but will lose information on seasonality for instance of some specific product at level 3 

Option 2: you work on level 3 and sum up to upper levels. Here you will lose any relationship between products and the impact of one product to another product. 

Option 3: you will work on level 2, do proportions for level 3 and sum up to level 1. 

Your choice. I hope you find this useful. Also you can ensemble some statsmodels algo and using it as regressors in Prophet to fine tune your prophet models using the above approach. 

So basically no easy answer here. 

Good luck

**Comment by bernardo-suez** [comment]
The three time series are summed, so I'm basically grouping (adding) them by calculating the total mean and variance for each time step, which would be your option two. I was wondering if there would be a way of getting the corresponding uncertainty interval with fewer assumptions, such as normality.

**Comment by tcuongd** [comment]
As Ruben said, it's pretty complicated. If you're looking for a rigorous way to combine prediction intervals, this might be a good reference: https://otexts.com/fpp3/rec-prob.html (note that I haven't tried it myself). You'll probably want to use the R packages created by Rob Hyndman for forecast reconciliation.

**Comment by bernardo-suez** [comment]
@tcuongd Thanks for the reference. I'll probably work again soon on a project in which I'll have to combine prediction intervals and will try the reference you send. I'll let you know how it went.
